src/main/java/pl/kzcwat/localincidentserver/region/RegionController.java

```diff
@@ -0,0 +1,80 @@
+package pl.kzcwat.localincidentserver.region;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import pl.kzcwat.localincidentserver.region.exception.RegionNotFoundException;
+import pl.kzcwat.localincidentserver.region.request.RegionRequest;
+
+import java.util.Optional;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/api/region")
+public class RegionController {
+
+    private final RegionService regionService;
+
+    @GetMapping()
+    public Iterable<RegionEntity> getAllRegions() {
+        return this.regionService.getAllRegions();
+    }
+
+    @GetMapping("/{regionId}")
+    public ResponseEntity<RegionEntity> getRegion(@PathVariable Long regionId) {
+        try {
+            Optional<RegionEntity> regionEntity = regionService.getRegion(regionId);
+            return regionEntity.map(value -> new ResponseEntity<>(value, HttpStatus.OK)).orElseGet(() -> new ResponseEntity<>(HttpStatus.NOT_FOUND));
+        } catch (IllegalArgumentException e) {
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        } catch (RegionNotFoundException e) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+    }
+
+    @PostMapping("/addRegion")
+    public ResponseEntity<RegionEntity> saveRegion(@RequestBody RegionRequest regionRequest) {
+        try {
+            return new ResponseEntity<>(regionService.saveRegion(regionRequest), HttpStatus.CREATED);
```

Comment:
```suggestion
            RegionEntity region = regionService.saveRegion(regionRequest);
            URI location = ServletUriComponentsBuilder
                         .fromCurrentRequest()
                         .path("/{id}")
                         .buildAndExpand(region.getId())
                         .toUri();
            return ResponseEntity.created(location).body(region).build();
```
Coś w tym stylu powinno być. Przy POST niech zwraca też lokalizację tego obiektu.